src/cpp/session/modules/SessionPackages.R

```diff
@@ -1159,3 +1171,95 @@ if (identical(as.character(Sys.info()["sysname"]), "Darwin") &&
    # delegate to 'setInternet2'
    utils::setInternet2(value)
 })
+
+.rs.addFunction("parseSecondaryReposIni", function(conf) {
+   lines <- readLines(conf)
+   repos <- list()
+
+   for (line in lines) {
+     part <- strsplit(line, "=")[[1]]
+     if (identical(length(part), 2L)) {
+        repo <- list(
+           name  = .rs.scalar(trimws(part[[1]])),
+           url = .rs.scalar(trimws(part[[2]])),
+           host = .rs.scalar("Custom"),
+           country = .rs.scalar("")
+        )
+
+        if (identical(tolower(as.character(repo$name)), "cran")) {
+           repo$name <- .rs.scalar("CRAN")
+           repos <- append(list(repo), repos, 1)
+        } else {
+           repos[[length(repos) + 1]] <- repo
+        }
+     }
+   }
+
+   repos
+})
+
+.rs.addFunction("parseSecondaryReposJson", function(conf) {
+   lines <- readLines(conf)
+   repos <- list()
+
+   entries <- .rs.fromJSON(paste(lines, collpse = "\n"))
+
+   for (entry in entries) {
+      url <- if (is.null(entry$url)) "" else url
+
+      repo <- list(
+         name  = .rs.scalar(entry$name),
+         url = .rs.scalar(url),
+         host = .rs.scalar("Custom"),
+         country = .rs.scalar("")
+      )
+
+      if (identical(tolower(as.character(repo$name)), "cran")) {
+         repo$name <- .rs.scalar("CRAN")
+         repos <- append(list(repo), repos, 1)
+      } else {
+         repos[[length(repos) + 1]] <- repo
+      }
+   }
+
+   repos
+})
+
+.rs.addFunction("getSecondaryRepos", function() {
+   conf <- ""
+
+   parser <- list(
+      ini = .rs.parseSecondaryReposIni,
+      json = .rs.parseSecondaryReposJson,
+      none = function(conf) list()
+   )
+   parserType <- "none"
+   
+   rCranReposUrl <- .Call("rs_getCranReposUrl")
+   if (nchar(rCranReposUrl) > 0) {
+      conf <- tempfile(fileext = ".conf")
+      
+      tryCatch({
+         download.file(rCranReposUrl, conf, extra = "-H 'Accept: text/javascript'")
+         parserType <- "json"
+      }, catch = function(e) {
+         ""
```

Comment:
Do you think we should log here if the error isn't caused by a mime type mismatch? (wondering if swallowing exceptions here might make it hart to diagnose network issues, cert issues, etc. that are preventing the URL from working)

Reply:
This code needs to be removed since RSPM just added support for `text/ini` files, needs to do some testing today and return the error.